circle.yml

```diff
@@ -0,0 +1,26 @@
+machine:
+  services:
+    - docker
+
+dependencies:
+  pre:
+      # set build status to pending on start
+    - ./util/docker/maketest.sh --pending ${GH_TOKEN}
+      # Login to docker
+    - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
+    - ./util/docker/baseimage/buildbaseimage.sh
+      # Actually the test step, but we want to cache it so it can go here.
+    - ./util/docker/circle/buildcircle.sh
```

Comment:
Does caching not work if we run this under `test`?  If possible, it'd be nice to have the main circleci status only go green if everything passes.